static_src/stores/route_store.js

```diff
@@ -40,6 +42,26 @@ class RouteStore extends BaseStore {
     return !!route.app_guid;
   }
 
+  getRouteURLForApp(app) {
+    const allRoutes = this.getAll();
+    const boundRoutes = allRoutes.filter((route) => route.app_guid === app.guid);
```

Comment:
I think it's because we set the `app_guid` when the request is made on the app page.